lib/_http_incoming.js

```diff
@@ -160,18 +160,27 @@ IncomingMessage.prototype._read = function _read(n) {
     readStart(this.socket);
 };
 
-
 // It's possible that the socket will be destroyed, and removed from
 // any messages, before ever calling this.  In that case, just skip
 // it, since something else is destroying this connection anyway.
-IncomingMessage.prototype.destroy = function destroy(error) {
-  // TODO(ronag): Implement in terms of _destroy
-  this.destroyed = true;
-  if (this.socket)
-    this.socket.destroy(error);
-  return this;
-};
+IncomingMessage.prototype._destroy = function _destroy(err, cb) {
+  if (!this.readableEnded || !this.complete) {
+    this.aborted = true;
+    this.emit('aborted');
+  }
 
+  // If aborted and the underlying socket not already destroyed,
+  // destroy it.
+  if (this.socket && !this.socket.destroyed && this.aborted) {
```

Comment:
I don't think you need to check socket.destroyed

Reply:
I agree, but I am having problems with this test:

parallel/test-http-client-spurious-aborted.js

Where it seems that the underlying socket is already destroyed when calling `IncomingMessage.destroy`. This causes `finished` to never call the callback.

Reply:
ah, `finished` should be able to handle that but maybe the `willEmitClose` check needs improvement.